kitsune/wiki/handlers.py

```diff
@@ -11,30 +12,41 @@ class DocumentListener(UserDeletionListener):
     """Listener for document-related tasks."""
 
     def on_user_deletion(self, user: User) -> None:
-        """Handle the deletion of a user."""
+        """Handle the deletion of a user.
 
-        documents = Document.objects.filter(contributors=user)
+        - Replace the user as a document contributor with content group members
+          if they were the only contributor.
+        - Anonymize the user's revision votes.
+        - Delete the user's non-approved revisions.
+        - Replace the user as creator/reviewer of approved revisions with the SUMO bot.
+        """
+        sumo_bot = Profile.get_sumo_bot()
         content_group = Group.objects.get(name=settings.SUMO_CONTENT_GROUP)
+
+        # Handle approved revisions first to avoid cascade deletion issues
+        Revision.objects.filter(creator=user, is_approved=True).update(creator=sumo_bot)
+
+        # Update reviewer and readied_for_localization_by fields
+        Revision.objects.filter(Q(reviewer=user) | Q(readied_for_localization_by=user)).update(
+            reviewer=sumo_bot, readied_for_localization_by=sumo_bot
+        )
```

Comment:
These have to be separate updates. There are many revisions where the `reviewer` field is non-NULL, but the `readied_for_localization_by` is NULL. So, in those cases, if the `reviewer` is the deleted user, the `readied_for_localization_by` field will be incorrectly updated to be the sumo bot as well, when it should have remained NULL.

Reply:
I think we should add another case to the `test_revision_reviewer_replacement` test that covers the above, either in this PR or I can add it in another PR. Something like:

```python
def test_revision_reviewer_replacement(self):
    ...
    # This revision has a reviewer, but no "readied_for_localization_by".
    rev4 = ApprovedRevisionFactory(creator=contributor, reviewer=self.user)
    ...
    self.assertEqual(rev4.creator.username, contributor.username)
    self.assertEqual(rev4.reviewer.username, settings.SUMO_BOT_USERNAME)
    # The "readied_for_localization_by" field should remain None.
    self.assertIsNone(rev4.readied_for_localization_by)
```